internal/service/domain/tree/tree.go

```diff
@@ -50,6 +53,42 @@ func handleError(err error) error {
 func (s *TreeService) Ready() bool {
 	return s.treeRepo != nil && s.sensorRepo != nil
 }
+func (s *TreeService) Create(ctx context.Context, treeCreate *entities.TreeCreate) (*entities.Tree, error) {
+	if treeCreate.PlantingYear == 0 {
+		return nil, handleError(errors.New("plantingYear cannot be null or zero"))
+	}
+	if treeCreate.Species == "" {
+		return nil, handleError(errors.New("species (Gattung) cannot be null or empty"))
+	}
+	if treeCreate.Number == "" {
+		return nil, handleError(errors.New("tree Number (Baum Nr) cannot be null or empty"))
+	}
+	if treeCreate.Latitude == 0 || treeCreate.Longitude == 0 {
+		return nil, handleError(errors.New("latitude and Longitude cannot be null or zero"))
+	}
+
+	fn := make([]entities.EntityFunc[entities.Tree], 0)
+	if treeCreate.TreeClusterID != nil {
+		treeClusterID, err := s.treeClusterRepo.GetByID(ctx, *treeCreate.TreeClusterID)
+		if err != nil {
+			return nil, handleError(err)
+		}
+		fn = append(fn, tree.WithTreeCluster(treeClusterID))
+	}
```

Comment:
If a tree is inserted into an existing tree cluster, the tree cluster coordinates must be updated